python/fusion_engine_client/analysis/analyzer.py

```diff
@@ -1335,59 +1335,57 @@ def plot_wheel_data(self):
-        self._plot_wheel_ticks_or_speeds(type='tick')
+        self._plot_wheel_ticks_or_speeds(source='wheel', type='speed')
+        self._plot_wheel_ticks_or_speeds(source='wheel', type='tick')
+        self._plot_wheel_ticks_or_speeds(source='vehicle', type='speed')
+        self._plot_wheel_ticks_or_speeds(source='vehicle', type='tick')
 
-    def _plot_wheel_ticks_or_speeds(self, type):
+    def _plot_wheel_ticks_or_speeds(self, source, type):
         """!
         @brief Plot wheel speed or tick data.
         """
         # Read the data. Try to determine which type of wheel output is present in the log (if any).
         if type == 'tick':
-            filename = 'wheel_ticks'
-            figure_title = 'Measurements: Wheel Encoder Ticks'
+            filename = '%s_ticks' % source
+            figure_title = 'Measurements: %s Encoder Ticks' % source.title()
 
-            wheel_measurement_type = self._auto_detect_message_type([RawWheelTickOutput, WheelTickInput])
-            vehicle_measurement_type = self._auto_detect_message_type([RawVehicleTickOutput, VehicleTickInput])
+            if source == 'wheel':
+                measurement_type = self._auto_detect_message_type([RawWheelTickOutput, WheelTickInput])
+            else:
+                measurement_type = self._auto_detect_message_type([RawVehicleTickOutput, VehicleTickInput])
 
             # Wheel ticks are raw (uncorrected) by definition.
-            raw_wheel_measurement_type = wheel_measurement_type
-            raw_vehicle_measurement_type = vehicle_measurement_type
+            raw_measurement_type = measurement_type
         else:
-            filename = 'wheel_speed'
-            figure_title = 'Measurements: Wheel Speed'
+            filename = '%s_speed' % source
+            figure_title = 'Measurements: %s Speed' % source.title()
 
-            wheel_measurement_type = self._auto_detect_message_type([WheelSpeedOutput, DeprecatedWheelSpeedMeasurement])
-            vehicle_measurement_type = self._auto_detect_message_type([VehicleSpeedOutput,
+            if source == 'wheel':
+                measurement_type = self._auto_detect_message_type([WheelSpeedOutput, DeprecatedWheelSpeedMeasurement])
+            else:
+                measurement_type = self._auto_detect_message_type([VehicleSpeedOutput,
                                                                        DeprecatedVehicleSpeedMeasurement])
 
-            raw_wheel_measurement_type = self._auto_detect_message_type([RawWheelSpeedOutput])
-            raw_vehicle_measurement_type = self._auto_detect_message_type([RawVehicleSpeedOutput])
+            if source == 'wheel':
+                raw_measurement_type = self._auto_detect_message_type([RawWheelSpeedOutput, WheelSpeedInput])
+            else:
+                raw_measurement_type = self._auto_detect_message_type([RawVehicleSpeedOutput, VehicleSpeedInput])
+
+            if measurement_type is None:
```

Comment:
Yeah admittedly this is a big odd/confusing. We originally did that to simplify stuff in a few places since we use the measurement type to set plot titles and such. It might be nice to just separate them completely, just need to do it carefully. I'll see how hard that would be.

In the second case with ticks, there's no such thing as corrected ticks so basically we always only have raw.

Reply:
Ok, wasn't so bad. I separated everything so we never do `measurement_type = raw_measurement_type`.